### PR TITLE
Enhance jump node handling

### PR DESCRIPTION
**Pull request by MjnMixael**
I set out to implement a simple jump node display name feature as requested in #1730 . I ended up enhancing handling of jump node in FRED across the board.

In addition to the new Jump Node Display Name (which allows XSTR), I implemented the following...

- Splits up the jump node dialog and the waypoint dialog into their own unique editor dialogs.
- Adds all the jump node features that were previously available by text editing the mission file to the new jump node dialog editor.
- Makes FRED display both the jump node Name and Display name in the FREDView instead of just "Jump Node".
- Moves the few instances of "subspacenode.pof" to a DEFINE in jumpnode.h. I thought about exposing that string in game_settings.tbl, but didn't see much utility in such a feature.
- Adds a set-jumpnode-display-name SEXP

### CODE REVIEW

File: fred2/fredrender.cpp
Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: No other object type does that kind of filtering for displayed values. I'm against doing it here.

Comment (Goober5000) [reply fred2/fredrender.cpp:546]: Well, in FRED, no other object type shows its display name in the info box, so that's really an argument towards not showing it at all.

What I'm saying is that in 99% of cases, the name and display name of a jump node will be identical.  When they're identical, showing the same string on two lines is redundant.

Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: Name and display name here are more like ship class and ship name. It makes sense to keep them separate. Don't approve the pr if you want to insist on such a tiny thing. Fine by me.

Comment (naomimyselfandi) [reply fred2/fredrender.cpp:546]: What are some use cases either way?

Comment (Goober5000) [reply fred2/fredrender.cpp:546]: Here are the three cases as I see them:

1. In the current form of the PR, High Noon would have the two jump nodes shown as follows:
```
Gamma Draconis 
Gamma Draconis
```
```
Vega
Vega
```
In this case, since the name and display name are identical, the extra line is redundant.  On the other hand, if you had a multi-jump mission with the same jump node in different areas, you might have
```
Capella#1
Capella
```
```
Capella#2
Capella
```
In that case, the extra line is not redundant.

2. My suggestion is to not display the extra line when the name and display name are the same.  If the two names are not the same, the extra line can be shown.

3. One point that came up in the above discussion is that no other object type shows its display name in the info box at all.  This is an argument toward not showing the display names for jump nodes regardless of whether they are the same or not.

Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: I'm not changing my code because this is merely about aesthetic preferences. It is extremely unlikely that you will change my opinion on this.

Having both lines no matter what signifies that there is additional data that could be used, even if they are currently the same. FRED is a designer first and foremost and should ere on displaying more information and being less automated as a general design principal. We're not Apple, we don't assume the FRED userbase is full of morons.

**Review by Goober5000 — APPROVED** [review]

File: fred2/fredrender.cpp
Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: Yes, I expect you'll go back and redo the whole thing in your preferred way like you do with a lot of people's PRs. You do know that makes you look like a giant douchebag, though, right?

In fact I expect you'll go back and change the code here to do it the way you want in your own follow-up PR. That's why you're approving it.

Comment (Goober5000) [reply fred2/fredrender.cpp:546]: You do realize that if this is the way you discuss the simplest suggestions, it makes people less likely to discuss complex ones?

Anyway, no, I don't plan to change the duplicate line behavior.  You've made your opinion on that point very clear.

Comment (Baezon) [reply fred2/fredrender.cpp:546]: Fixes or addendums to recent changes are very common, and completely normal. Please do keep in mind that all PRs need approvals from other SCP members, even Goober's.

Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: Goober goes out of his way to nitpick my PRs, more than most based on my observations (whether he'll admit that or not). And he often goes back and just rewrites whole sections just for personal style and functionality or bugs. I would rather have my PRs sit for years than have him review them, which I suspect he knows and I suspect that's why he often blocks them for super minor things; to frustrate me, given our relational issues of the past. I'm sure he'll respond and play all innocent. But seriously ya'll should see the correspondence we've had. There's a reason I have blocked him on every single platform he knows to contact me on. I, unfortunately, have no method to do that on GitHub in any meaningful way or I would.

Goober is the sole reason I stopped working on any code I don't need for my personal projects or wasn't directly requested of me via PMs.

Comment (Goober5000) [reply fred2/fredrender.cpp:546]: I don't review PRs for the purpose of frustrating you.  As for the rest, I direct you to Matthew 5:23-24 and 1 John 3:14.

Comment (MjnMixael) [reply fred2/fredrender.cpp:546]: Goober, you accused me of not being Christian _at all_, so I don't think that applies according to your own belief system. I'm not one of your "brothers" bruh.

Comment (Goober5000) [reply fred2/fredrender.cpp:546]: > Goober, you accused me of not being Christian _at all_

That doesn't sound right.  I have to assume that was a miscommunication or misinterpretation.

> so I don't think that applies according to your own belief system

Does it apply according to yours?

Comment (Baezon) [reply fred2/fredrender.cpp:546]: Guys, this is not the place for this. The issue has been resolved, and the PR approved.

**Pull request merged**
